src/ejabberd_c2s.erl

```diff
@@ -1065,12 +1054,10 @@ handle_info(system_shutdown, StateName, StateData) ->
     case StateName of
         wait_for_stream ->
             send_header(StateData, ?MYNAME, <<"1.0">>, <<"en">>),
-            send_element_from_server_jid(StateData, mongoose_xmpp_errors:system_shutdown()),
-            send_trailer(StateData),
+            c2s_stream_error(mongoose_xmpp_errors:system_shutdown(), StateData),
             ok;
         _ ->
-            send_element_from_server_jid(StateData, mongoose_xmpp_errors:system_shutdown()),
-            send_trailer(StateData),
+            c2s_stream_error(mongoose_xmpp_errors:system_shutdown(), StateData),
             ok
     end,
     {stop, normal, StateData};
```

Comment:
In this case of, regardless of the branch, we'll return `{stop, normal, StateData}`, which is precisely what `c2s_stream_error` returns, so you could just
```erlang
    case StateName of
        wait_for_stream ->
            send_header(StateData, ?MYNAME, <<"1.0">>, <<"en">>),
            c2s_stream_error(mongoose_xmpp_errors:system_shutdown(), StateData);
        _ ->
            c2s_stream_error(mongoose_xmpp_errors:system_shutdown(), StateData),
    end;
```

Reply:
What the heck, or even shorter:
```erlang
    case StateName of
        wait_for_stream -> send_header(StateData, ?MYNAME, <<"1.0">>, <<"en">>);
        _ -> ok
    end,
    c2s_stream_error(mongoose_xmpp_errors:system_shutdown(), StateData);
```
or also
```erlang
handle_info(system_shutdown, wait_for_stream, StateData) ->
    send_header(StateData, ?MYNAME, <<"1.0">>, <<"en">>);
    c2s_stream_error(mongoose_xmpp_errors:system_shutdown(), StateData);
handle_info(system_shutdown, _StateName, StateData) ->
    c2s_stream_error(mongoose_xmpp_errors:system_shutdown(), StateData);
```
Not sure which one is better, maybe the first, but it's mostly just opinion here 😛